### PR TITLE
Feature/role based access controls#16

### PR DESCRIPTION
**Pull request by iowusu**

### CODE REVIEW

File: state-conductor-modules/src/main/ml-config/security/roles/state-conductor-job-writer-role.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "role-name": "state-conductor-job-writer-role",
+  "description": "Permits reading flow documents and knowing flow status",
+  "role": [
+    "state-conductor-reader-role"
+  ],
+  "privilege": [
+    {
+      "privilege-name": "state-conductor-execute",
+      "action": "http://marklogic.com/state-conductor/execute",
```
Comment (aclavio) [inline state-conductor-modules/src/main/ml-config/security/roles/state-conductor-job-writer-role.json]: The privilege name in the `xdmp.securityAssert() `function calls is `http://marklogic.com/xdmp/privileges/state-conductor/excute` shouldn't these match?  (Also note `execute` is missing an "e")

File: state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs [hunk 2]
```diff
@@ -1097,6 +1132,7 @@ function hasScheduleElapsed(context, now) {
  * @returns
  */
 function getJobDocuments(options) {
+
   const count = options.count || 100;
   const flowStatus = Array.isArray(options.flowStatus) ? options.flowStatus : [FLOW_STATUS_NEW, FLOW_STATUS_WORKING];
   const flowNames = Array.isArray(options.flowNames) ? options.flowNames : [];
```
Comment (aclavio) [inline state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs]: Should we also protect these?:
* emmitEvent
* createStateConductorJob

Comment (freshie) [reply state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs]: emmitEvent calls resume waiting so that might be ok.

Comment (iowusu) [reply state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs]: createStateConductorJob is now protected.

**Comment by iowusu** [comment]
I have resolved the about issues.

**Pull request merged**